reconcile/database_access_manager.py

```diff
@@ -0,0 +1,506 @@
+        )
+        generator = DBScriptGenerator(db_access=db_access, engine=engine)
+        print(generator.generate_script())
+        # create script secret
+        script_secret_name = f"{resource_name}-script"
+        managed_resources.append(
+            DBAMResource(
+                resource=generate_script_secret_spec(
+                    script_secret_name,
+                    generator.generate_script(),
+                ),
+                clean_up=True,
+            )
+        )
+        # create user secret
+        managed_resources.append(
+            DBAMResource(
+                resource=generate_user_secret_spec(resource_name, db_access),
+                clean_up=True,
+            )
+        )
+        # create pull secret
+        labels = pull_secret["labels"] or {}
+        pull_secret_resources = orb.fetch_provider_vault_secret(
+            path=pull_secret["path"],
+            version=pull_secret["version"],
+            name=f"{resource_name}-pull-secret",
+            labels=labels,
+            annotations=pull_secret["annotations"] or {},
+            type=pull_secret["type"],
+            integration=QONTRACT_INTEGRATION,
+            integration_version=QONTRACT_INTEGRATION_VERSION,
+            settings=settings,
+        )
+        managed_resources.append(
+            DBAMResource(resource=pull_secret_resources, clean_up=True)
+        )
+        # create job
+        managed_resources.append(
+            DBAMResource(
+                resource=get_job_spec(
+                    JobData(
+                        engine=engine,
+                        engine_version="15.4-alpine",
+                        name_suffix=db_access.name,
+                        image_repository=image_repository,
+                        service_account_name=resource_name,
+                        rds_admin_secret_name=resource.output_resource_name,
+                        script_secret_name=script_secret_name,
+                        pull_secret=f"{resource_name}-pull-secret",
+                    )
+                ),
+                clean_up=False,
+            )
+        )
+
+        return managed_resources
+
+    def run(self, dry_run: bool) -> None:
+        settings = queries.get_app_interface_settings()
+
+        sql_query_settings = settings.get("sqlQuery")
+        if not sql_query_settings:
+            raise KeyError("sqlQuery settings are required")
+
+        image_repository = sql_query_settings["imageRepository"]
+        pull_secret = sql_query_settings["pullSecret"]
+
+        encounteredErrors = False
+
+        for namespace in get_database_access_namespaces():
+            with OC_Map(
+                namespaces=[namespace.dict(by_alias=True)],
+                integration=QONTRACT_INTEGRATION,
+                settings=settings,
+            ) as oc_map:
+                for external_resource in namespace.external_resources:
+                    for resource in external_resource.resources:
```

Comment:
is there no filtering here to only look at RDS resources?

Reply:
No, filtering for nonUnique attributes is not possible as of now